docs/sources/developers/plugins/migration-guide.md

```diff
@@ -0,0 +1,151 @@
++++
+title = "Migrate a plugin"
+type = "docs"
++++
+
+# Migrate a plugin
+
+This guide is aimed towards plugin authors who want to migrate their plugins to Grafana 7.0, and users who are using custom plugins and want to know whether it’s safe to upgrade.
+
+## What's new in Grafana 7?
+
+Grafana 7 introduced a whole new plugin platform based on React that supersedes the previous Angular based platform. Plugins built using Angular will still work for the foreseeable future, though we encourage new plugin authors to develop against the new platform.
+
+### New data format
+
+Along with the move to React, the new plugin platform introduced a new internal data format, called [data frames](https://grafana.com/docs/grafana/latest/developers/plugins/data-frames/#data-frames). Previously, data source plugins could send data either as time series or tables. With data frames, data sources can send any type of data in a table-like structure. This gives you more flexibility for data you want to visualize in Grafana.
+
+### Improved TypeScript support
+
+While the previous Angular-based plugin SDK did support TypeScript, for the React platform, we’ve bet hard on TypeScript. All our APIs are now TypeScript, which might require existing code to update to the new stricter requirements. Grafana 7 also introduced several new APIs for plugin developers that take advantage of many of the new features in Grafana 7.
+
+### Grafana Toolkit
+
+With Grafana 7.0, we released a new tool for making it easier to develop plugins. Before, you’d use Gulp, Grunt, or similar tools to generate the minified assets.
+
+### Field options
+
+Grafana 7 introduced the concept of [_field options_](https://grafana.com/docs/grafana/latest/panels/field-configuration-options/#field-configuration-options), a new way of configuring your data before it gets visualized. Since this was not available in previous versions, any plugin that enabled field-based configuration will not work in previous versions of Grafana.
```

Comment:
```suggestion
Grafana 7 introduced the concept of [_field options_](https://grafana.com/docs/grafana/latest/panels/field-configuration-options/#field-configuration-options), a new way of configuring your data before it gets visualized. Since this was not available in previous versions, any plugin that enables field-based configuration will not work in previous versions of Grafana.
```